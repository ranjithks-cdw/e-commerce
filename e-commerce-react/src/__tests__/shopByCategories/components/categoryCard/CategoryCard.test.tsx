import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import CategoryCard from '../../../../features/shopByCategories/components/categoryCard/CategoryCard';

const data = {
    categoryName: "Title",
    categoryImageUrl: "",
};

describe('CategoryCard', () => {
    it("renders component", async () => {
        render(<CategoryCard {...data} />);

        expect(await screen.findByText(data.categoryName)).toBeInTheDocument();

        const image = screen.getByAltText(data.categoryName);
        expect(image).not.toBeNull();
        expect(image).toHaveAttribute('src', data.categoryImageUrl);
    })

    it("matches snapshot", () => {
        const container = renderer.create(<CategoryCard {...data} />).toJSON;
        expect(container).toMatchSnapshot();
    })
})