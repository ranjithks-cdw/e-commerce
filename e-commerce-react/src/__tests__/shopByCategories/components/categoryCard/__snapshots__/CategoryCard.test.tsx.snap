// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryCard matches snapshot 1`] = `[Function]`;
